admin_manual/installation/nginx.rst

```diff
@@ -114,75 +69,112 @@ webroot of your nginx installation. In this example it is
       gzip_proxied expired no-cache no-store private no_last_modified no_etag auth;
       gzip_types application/atom+xml application/javascript application/json application/ld+json application/manifest+json application/rss+xml application/vnd.geo+json application/vnd.ms-fontobject application/x-font-ttf application/x-web-app-manifest+json application/xhtml+xml application/xml font/opentype image/bmp image/svg+xml image/x-icon text/cache-manifest text/css text/plain text/vcard text/vnd.rim.location.xloc text/vtt text/x-component text/x-cross-domain-policy;
 
-      # Uncomment if your server is build with the ngx_pagespeed module
-      # This module is currently not supported.
+      # Pagespeed is not supported by Nextcloud, so if your server is built
+      # with the `ngx_pagespeed` module, uncomment this line to disable it.
       #pagespeed off;
-
-      location / {
-          rewrite ^ /index.php;
+      
+      # HTTP response headers borrowed from Nextcloud `.htaccess`
+      add_header Referrer-Policy                      "no-referrer"   always;
+      add_header X-Content-Type-Options               "nosniff"       always;
+      add_header X-Download-Options                   "noopen"        always;
+      add_header X-Frame-Options                      "SAMEORIGIN"    always;
+      add_header X-Permitted-Cross-Domain-Policies    "none"          always;
+      add_header X-Robots-Tag                         "none"          always;
+      add_header X-XSS-Protection                     "1; mode=block" always;
+      
+      # Remove X-Powered-By, which is an information leak
+      fastcgi_hide_header X-Powered-By;
+      
+      # Path to the root of your installation
+      root /var/www/nextcloud;
+      
+      # Specify how to handle directories -- specifying `/index.php$request_uri`
+      # here as the fallback means that Nginx always exhibits the desired behaviour
+      # when a client requests a path that corresponds to a directory that exists
+      # on the server. In particular, if that directory contains an index.php file,
+      # that file is correctly served; if it doesn't, then the request is passed to
+      # the front-end controller. This consistent behaviour means that we don't need
+      # to specify custom rules for certain paths (e.g. images and other assets,
+      # `/updater`, `/ocm-provider`, `/ocs-provider`), and thus
+      # `try_files $uri $uri/ /index.php$request_uri`
+      # always provides the desired behaviour.
+      index index.php index.html /index.php$request_uri;
+      
+      # Default Cache-Control policy
+      expires 1m;
```

Comment:
This seems to cause update delays for dynamic elements requested via APIs.
For example when using the news plugin, it causes feed items which I've marked as read still being displayed in the list for 60 seconds. Probably it also affects other elements and plugins as well.